src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -1026,7 +1041,7 @@ def populate_column_dependency(
     elif dependent_column_names == COLUMN_DEPENDENCY_ALL:
         column_states[quoted_c_name].depend_on_same_level = True
         column_states.columns_referencing_all_columns.add(quoted_c_name)
-    else:
```

Comment:
Added, although it shouldn't be reachable.

Reply:
lol no it's extremely reachable since the set can just be empty. Removed the else clause. The goal here is just to concisely tell the type checker that `dependent_column_names` isn't None.

Reply:
Can we do `elif dependent_column_names is not None` then add a final unreachable `else`? I'm a bit paranoid to not have a "default failure" when there is a lot of if/elif involved.